homeassistant/components/androidtv_remote/config_flow.py

```diff
@@ -0,0 +1,172 @@
+                await self.async_set_unique_id(format_mac(self.mac))
+                self._abort_if_unique_id_configured(updates={CONF_HOST: self.host})
+                return await self._async_start_pair()
+            except (CannotConnect, ConnectionClosed):
+                errors["base"] = "cannot_connect"
+        return self.async_show_form(
+            step_id="user",
+            data_schema=STEP_USER_DATA_SCHEMA,
+            errors=errors,
+        )
+
+    async def _async_start_pair(self) -> FlowResult:
+        """Start pairing with the Android TV. Navigate to the pair flow to enter the PIN shown on screen."""
+        assert self.host
+        self.api = create_api(self.hass, self.host)
+        self.api.generate_cert_if_missing()
+        await self.api.async_start_pairing()
+        return await self.async_step_pair()
+
+    async def async_step_pair(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the pair step."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            try:
+                pin = user_input["pin"]
+                assert self.api
+                await self.api.async_finish_pairing(pin)
+                if self.reauth_entry:
+                    await self.hass.config_entries.async_reload(
+                        self.reauth_entry.entry_id
+                    )
+                    return self.async_abort(reason="reauth_successful")
+                assert self.name
+                return self.async_create_entry(
+                    title=self.name,
+                    data={
+                        CONF_HOST: self.host,
+                        CONF_NAME: self.name,
+                        CONF_MAC: self.mac,
+                    },
+                )
+            except InvalidAuth:
+                errors["base"] = "invalid_auth"
+            except ConnectionClosed:
+                try:
+                    return await self._async_start_pair()
+                except (CannotConnect, ConnectionClosed):
+                    return self.async_abort(reason="cannot_connect")
+        return self.async_show_form(
+            step_id="pair",
+            data_schema=STEP_PAIR_DATA_SCHEMA,
+            description_placeholders={CONF_NAME: self.name},
+            errors=errors,
+        )
+
+    async def async_step_zeroconf(
+        self, discovery_info: zeroconf.ZeroconfServiceInfo
+    ) -> FlowResult:
+        """Handle zeroconf discovery."""
+        self.host = discovery_info.host
+        self.name = discovery_info.name.removesuffix("._androidtvremote2._tcp.local.")
+        self.mac = discovery_info.properties.get("bt")
+        assert self.mac
+        await self.async_set_unique_id(format_mac(self.mac))
+        self._abort_if_unique_id_configured(
+            updates={CONF_HOST: self.host, CONF_NAME: self.name}
+        )
+        self.context.update({"title_placeholders": {CONF_NAME: self.name}})
```

Comment:
"title_placeholders"?

Reply:
Yes this is the title of the discovery card https://user-images.githubusercontent.com/9987465/226156769-7eae4cbd-307e-4bcb-bae7-d94ea614d811.png